hyperon_das_atomdb/adapters/redis_mongo_db.py

```diff
@@ -87,31 +94,68 @@ class MongoDBIndex(Index):
     def __init__(self, collection: Collection) -> None:
         self.collection = collection
 
-    def create(self, atom_type: str, field: str, **kwargs) -> Tuple[str, Any]:
+    def create(
+        self,
+        atom_type: str,
+        field: str,
+        fields: Optional[List[str]] = None,
+        index_type: Optional[MongoIndexType] = None,
+        **kwargs,
+    ) -> Tuple[str, Any]:
         conditionals = {}
 
         for key, value in kwargs.items():
             conditionals = {key: {"$eq": value}}
             break  # only one key-value pair
 
-        index_id = f"{atom_type}_{self.generate_index_id(field, conditionals)}"
-
+        if fields:
+            index_id = (
+                f"{atom_type}_{self.generate_index_id(','.join(fields), conditionals)}" +
+                (f"_{index_type.value}" if index_type else "")
+            )
+        else:
+            index_id = (
+                f"{atom_type}_{self.generate_index_id(field, conditionals)}" +
+                (f'_{index_type.value}' if index_type else "")
+            )
+        index_type: MongoIndexType = (
+            index_type
+            or
+            (MongoIndexType.COMPOUND if fields is not None else MongoIndexType.FIELD)
+        )
+        index_props = {
+            'index_type': index_type, 'conditionals': conditionals, 
+            'index_name': index_id, 'fields': fields or [field]
+            }
         index_conditionals = {"name": index_id}
-
+ 
         if conditionals:
-            index_conditionals["partialFilterExpression"] = conditionals
+            index_conditionals["partialFilterExpression"] = index_props['conditionals']
+
+        if index_type == MongoIndexType.TEXT:
+            index_list = (
+                [((f, 'text') for f in fields)]
```

Comment:
It seems that there is an extra pair of `(` and `)`.

```suggestion
                [(f, 'text') for f in fields]
```